### PR TITLE
Field textinput pixel perfect input

### PR DESCRIPTION
**Pull request by samelhusseini**
## The basics
- [x] I branched from develop
- [x] My pull request is against develop
- [x] My code follows the [style guide](https://developers.google.com/blockly/guides/modify/web/style-guide)

### Resolves

Pixel perfect input.
Related to https://github.com/google/blockly/issues/2694. We still need to know the font height.

### Proposed Changes

Fix sizing of text input field in zelos. 
Fix sizing issues in other browsers. 

### Reason for Changes

Rendering.

### Test Coverage

Tested the playground with both geras, thrasos and zelos.

Tested on:
* Desktop Chrome
* Desktop Firefox
* Desktop Safari

### CODE REVIEW

File: core/css.js [hunk 1]
```diff
@@ -360,7 +360,7 @@ Blockly.Css.CONTENT = [
     'fill: #000;',
   '}',
 
-  '.blocklyEditableText:hover>rect {',
+  '.blocklyEditableText:not(.editing):hover>rect {',
```
Comment (samelhusseini) [inline core/css.js]: Firefox, Safari include the stroke when calculating sizing info. This ensures that when we calculate the size, there's not stroke.

File: core/field_textinput.js [hunk 2]
```diff
@@ -638,21 +619,15 @@ Blockly.FieldTextInput.prototype.getValueFromEditorText_ = function(text) {
  */
 Blockly.FieldTextInput.prototype.getScaledBBox = function() {
   if (this.fullBlockClickTarget_) {
-    var heightWidth = this.sourceBlock_.getHeightWidth();
+    var xy = this.getClickTarget_().getBoundingClientRect();
   } else {
-    var heightWidth = this.borderRect_.getBBox();
+    var xy = this.borderRect_.getBoundingClientRect();
```
Comment (samelhusseini) [inline core/field_textinput.js]: The trouble with the previous method (bbox + absoluteXY), is that absolute XY gets the size info of the click target / field svg root. This includes both the border rect and the text element. The text element (currently in geras) is slightly larger than the border rect which introduces the discrepancy when calculating positioning. The goal of sizing the text input is to match the dimensions of the border rect, hence why I'm switching it here to only use those dimensions.

File: tests/mocha/field_angle_test.js [hunk 3]
```diff
@@ -201,10 +201,14 @@ suite('Angle Fields', function() {
       this.angleField.htmlInput_ = Object.create(null);
       this.angleField.htmlInput_.oldValue_ = '1';
       this.angleField.htmlInput_.untypedDefaultValue_ = 1;
+      this.stub = sinon.stub(this.angleField, 'resizeEditor_');
```
Comment (rachel-fenichel) [inline tests/mocha/field_angle_test.js]: Why do you need to stub this?  The FieldAngle shouldn't be rendered, so resizeEditor shouldn't be called.

Comment (samelhusseini) [reply tests/mocha/field_angle_test.js]: ``onHtmlInputChange_`` is called, which in turn now calls ``resizeEditor_``

File: core/field_textinput.js (hunk 2 above)
Comment (rachel-fenichel) [reply core/field_textinput.js]: Cool.  Removing the workarounds certainly seems positive to me.

**Pull request merged**